### PR TITLE
Update changelog for the next release

### PR DESCRIPTION
**Pull request by josh146**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -84,45 +137,110 @@
...
+  ...    qml.RY(y, wires=[1])
+  ...    qml.CNOT(wires=[0, 1])
+  ...    return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+  >>> for _ in range(10):
+  ...    circuit(0.432, 0.12)
+  >>> print(dev.num_executions)
+  10
+  ```
+
 <h3>Improvements</h3>
```
Comment (josh146) [inline .github/CHANGELOG.md:197]: I tried to order the improvements section from most useful to users (improvements that are user-facing or even verging on new user-facing features), down to least useful (improvements that are implementation details).

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -374,8 +412,9 @@
 
 This release contains contributions from (in alphabetical order):
 
-Thomas Bromley, Christina Lee, Olivia Di Matteo, Anthony Hayes, Theodor Isacsson, Josh Izaac,
-Nathan Killoran, Shumpei Kobayashi, Romain Moyard, Maria Schuld, Antal Száva.
+Juan Miguel Arrazola, Thomas Bromley, Christina Lee, Alain Delgado Gran, Olivia Di Matteo, Anthony
+Hayes, Theodor Isacsson, Josh Izaac, Soran Jahangiri, Nathan Killoran, Shumpei Kobayashi, Romain
+Moyard, Zeyue Niu, Maria Schuld, Antal Száva.
```
Comment (josh146) [inline .github/CHANGELOG.md:417]: In included qchem contributors here as well (they are also included in the qchem changelog)

Comment (josh146) [reply .github/CHANGELOG.md:417]: Let me know if I'm missing anyone!

**Comment by josh146** [comment]
Rendered version here: https://pennylane--932.org.readthedocs.build/en/932/development/release_notes.html

File: qchem/CHANGELOG.md [hunk 3]
```diff
@@ -1,38 +1,30 @@
...
 
-  - The functions ``observable``, ``one_particle`` and ``two_particle`` are
+  - The functions `observable`, `one_particle` and `two_particle` are
     significantly simplified.
 
   - The methodology to build many-body observables in PL-QChem is more consistent.
 
-  - Users do not need to keep track of the contribution due to core orbitals
+  - There is no longer a need to keep track of the contribution due to core orbitals
     when an active space is defined. This is now handle internally.
```
Comment (co9olguy) [inline qchem/CHANGELOG.md:20]:
```suggestion
    when an active space is defined. This is now handled internally.
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -37,37 +64,63 @@
...
+  automatically make use of the new batch API---providing a speedup during optimization.
+
+* Gradient recipes are now much more powerful, allowing for operations to define their gradient
+  via an arbitrary linear combination of circuit evaluations.
+  [(#909)](https://github.com/PennyLaneAI/pennylane/pull/909)
+  [(#915)](https://github.com/PennyLaneAI/pennylane/pull/915)
+
+  With this change, gradient recipes can now be of the form
+  :math:`\frac{\partial}{\partial\phi_k}f(\phi_k) = \sum_{i} c_i f(a_i \phi_k + s_i )`,
+  and are no longer restricted to two-term shifts with identical (but negated) shift values.
```
Comment (co9olguy) [inline .github/CHANGELOG.md:86]:
```suggestion
  and are no longer restricted to two-term shifts with identical (but opposite in sign) shift values.
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -84,45 +137,110 @@
       return qml.density_matrix(wires=[1])  # wire 0 is traced out
   ```
 
-* The controlled rotation operations `CRX`, `CRY`, `CRZ`, and `CRot`, now have
-  parameter-shift gradients defined, allowing for analytic gradients on hardware
-  without decomposition.
-  [(#915)](https://github.com/PennyLaneAI/pennylane/pull/915)
-  
+* Adds the square root X gate `SX`. [(#871)](https://github.com/PennyLaneAI/pennylane/pull/871)
```
Comment (co9olguy) [inline .github/CHANGELOG.md:140]:
```suggestion
* Adds the square-root X gate `SX`. [(#871)](https://github.com/PennyLaneAI/pennylane/pull/871)
```

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -84,45 +137,110 @@
...
+  @qml.qnode(dev)
+  def circuit():
+      qml.SX(wires=[0])
+      return qml.expval(qml.PauliZ(wires=[0]))
+  ```
+
+* Two new hardware-efficient particle-conserving templates have been implemented
+  to perform VQE-based quantum chemistry simulations. The new templates apply
+  several layers of the particle-conserving entanglers proposed in Figs. 2a and 2b
+  of the article by Barkoutsos *et al*. in [arXiv:1805.04340](https://arxiv.org/abs/1805.04340)
```
Comment (co9olguy) [inline .github/CHANGELOG.md:154]:
```suggestion
  of Barkoutsos *et al*., [arXiv:1805.04340](https://arxiv.org/abs/1805.04340)
```

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -84,45 +137,110 @@
...
+  [(#875)](https://github.com/PennyLaneAI/pennylane/pull/875)
+  [(#876)](https://github.com/PennyLaneAI/pennylane/pull/876)
+
+<h4>Estimate and track resources</h4>
+
+* The `QuantumTape` class now contains basic resource estimation functionality. The method
+  `tape.get_resources()` returns a dictionary with a list of the constituent operations and the
+  number of times they were run. Similarly, `tape.get_depth()` computes the circuit depth.
```
Comment (co9olguy) [inline .github/CHANGELOG.md:162]:
```suggestion
  number of times they appear in the circuit. Similarly, `tape.get_depth()` computes the circuit depth.
```

File: .github/CHANGELOG.md [hunk 8]
```diff
@@ -84,45 +137,110 @@
...
+  loader will attempt to refresh devices if the required plugin device cannot be found.
+  This will result in an improved experience if installing PennyLane and plugins within
+  a running Python session (for example, on Google Colab), and avoid the need to
+  restart the kernel/runtime.
+  [(#907)](https://github.com/PennyLaneAI/pennylane/pull/907)
+
+* When using `grad_fn = qml.grad(cost)` to compute the gradient of a cost function with the Autograd
+  interface, the value of the intermediate forward pass is now available via the `grad_fn.forward`
+  property:
+  [(#914)](https://github.com/PennyLaneAI/pennylane/pull/914)
```
Comment (co9olguy) [inline .github/CHANGELOG.md:223]:
```suggestion
  property
  [(#914)](https://github.com/PennyLaneAI/pennylane/pull/914):
```

File: .github/CHANGELOG.md [hunk 9]
```diff
@@ -84,45 +137,110 @@
...
 
   ```python
-  qml.enable_tape()
+  def cost_fn(x, y):
+      return 2*np.sin(x[0])*np.exp(-x[1]) + x[0]**3 + np.cos(y)
```
Comment (co9olguy) [inline .github/CHANGELOG.md:227]:
```suggestion
      return 2 * np.sin(x[0]) * np.exp(-x[1]) + x[0] ** 3 + np.cos(y)
```

File: .github/CHANGELOG.md [hunk 10]
```diff
@@ -84,45 +137,110 @@
...
+  [(#916)](https://github.com/PennyLaneAI/pennylane/pull/916)
 
   ```pycon
-  >>> f(0.4)
-  tensor([0.89431013, 0.9510565 ], requires_grad=True)
+  >>> opt = qml.GradientDescentOptimizer()
+  >>> params, cost = opt.step_and_cost(cost_fn, params)
   ```
 
 * PennyLane provides a new, experimental module `qml.proc` which provides framework agnostic
```
Comment (co9olguy) [inline .github/CHANGELOG.md:246]:
```suggestion
* PennyLane provides a new experimental module `qml.proc` which provides framework agnostic
```

File: .github/CHANGELOG.md [hunk 11]
```diff
@@ -231,6 +289,30 @@
...
+* The MultiRZ gate now has a defined generator, allowing it to be used in quantum natural gradient
+  optimization.
+  [(#912)](https://github.com/PennyLaneAI/pennylane/pull/912)
+
+* The CRot gate now has a `decomposition` method, which breaks the gate down into rotations
+  and CNOT gates. This allows `CRot` to be used on devices that do not natively support it.
+  [(#908)](https://github.com/PennyLaneAI/pennylane/pull/908)
+
+* The classical processing in the `MottonenStatePreparation` template has been largely
+  rewritten to use dense matrices and tensor manipulations where ever possible.
```
Comment (co9olguy) [inline .github/CHANGELOG.md:301]:
```suggestion
  rewritten to use dense matrices and tensor manipulations wherever possible.
```

File: .github/CHANGELOG.md [hunk 12]
```diff
@@ -231,6 +289,30 @@
...
+  This is a preparation to support differentiation through the template in future.
```
Comment (co9olguy) [inline .github/CHANGELOG.md:302]:
```suggestion
  This is in preparation to support differentiation through the template in the future.
```

File: .github/CHANGELOG.md [hunk 13]
```diff
@@ -231,6 +289,30 @@
...
+  [(#851)](https://github.com/PennyLaneAI/pennylane/pull/851)
+
+  The `cache` argument should be an integer specifying the size of the cache. For example, a
+  cache of size 10 is created using:
+
+  ```pycon
+  >>> dev = qml.device("default.qubit", wires=2, cache=10)
+  ```
+
 * The `Operation`, `Tensor`, and `MeasurementProcess` classes now has the `__copy__` special method
```
Comment (co9olguy) [inline .github/CHANGELOG.md:316]:
```suggestion
* The `Operation`, `Tensor`, and `MeasurementProcess` classes now have the `__copy__` special method
```

File: .github/CHANGELOG.md [hunk 14]
```diff
@@ -231,6 +289,30 @@
...
   defined. This allows us to ensure that, when a shallow copy is performed of an operation, the
   mutable list storing the operation parameters is *also* shallow copied. Both the old operation and
```
Comment (co9olguy) [inline .github/CHANGELOG.md:318]: Can't suggest, but just below, would be tood to move the PR link elsewhere. Looks a bit awkward to have a comma, then a bracket, then a code block

File: .github/CHANGELOG.md [hunk 15]
```diff
@@ -328,8 +368,6 @@
   beyond VQE. Use of `VQECost` is still possible but will result in a deprecation warning.
   [(#913)](https://github.com/PennyLaneAI/pennylane/pull/913)
 
-<h3>Documentation</h3>
-
 <h3>Bug fixes</h3>
 
 * The `default.qubit.tf` device is updated to handle TensorFlow objects (e.g.,
```
Comment (co9olguy) [inline .github/CHANGELOG.md:373]: The inverse tricks aren't really "Bug fixes". More "Improvements"?

**Review by co9olguy — APPROVED** [review]
Amazing! :tada:

File: .github/CHANGELOG.md (hunk 15 above)
Comment (josh146) [reply .github/CHANGELOG.md:373]: This one?

> The `default.qubit.tf` device is updated to handle TensorFlow objects (e.g., `tf.Variable`) as gate parameters correctly when using the `MultiRZ` and `CRot` operations.

Previously, this would error if you passed a tensorflow variable as the gate parameter, so I would class this one as a bug fix 🤔

Comment (co9olguy) [reply .github/CHANGELOG.md:373]: no, the one with the word "tricks" in it :)

File: .github/CHANGELOG.md (hunk 14 above)
Comment (josh146) [reply .github/CHANGELOG.md:318]: done

File: .github/CHANGELOG.md (hunk 15 above)
Comment (josh146) [reply .github/CHANGELOG.md:374]: Oh! Right.

I think this one is badly worded. It was a bug fix, in that we _thought_ inverses were being applied via the permutation methods, but there was an error in the if statement, causing them to fallback to the einsum `_apply_unitary`.

Comment (josh146) [reply .github/CHANGELOG.md:374]: Reworded this one!
